processing/src/main/java/org/apache/druid/segment/QueryableIndexSegment.java

```diff
@@ -81,6 +81,6 @@ public <T> T as(@Nonnull Class<T> clazz)
       return (T) new QueryableIndexRowsAndColumns(index);
     }
 
-    return null;
+    return Segment.super.as(clazz);
```

Comment:
Is this change related to the PR ?

Reply:
Yes there were some missing calls to the default as, which were fixed